pkg/model/application_live_state.go

```diff
@@ -144,3 +144,27 @@ func (s *ApplicationLiveStateSnapshot) determineLambdaAppHealthStatus() {
 	}
 	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
 }
+
+// DetermineApplicationHealthStatus updates the health status of the application based on the health status of its resources.
+func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
+	app := s.ApplicationLiveState
+	if app == nil {
+		return
+	}
+
+	for _, r := range app.Resources {
+		if r.HealthStatus == ResourceState_UNHEALTHY {
+			s.HealthStatus = ApplicationLiveStateSnapshot_UNHEALTHY
+			return
+		}
+	}
+
+	for _, r := range app.Resources {
+		if r.HealthStatus == ResourceState_UNKNOWN {
+			s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
+			return
+		}
+	}
+
+	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
+}
```

Comment:
```suggestion
func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
	app := s.ApplicationLiveState
	if app == nil {
		return
	}
	
	unhealthy := false
	unknown := false

	for _, r := range app.Resources {
		if r.HealthStatus == ResourceState_UNHEALTHY {
			unhealthy = true
		}
		if r.HealthStatus == ResourceState_UNKNOWN {
			unknown = true
		}
	}

	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY
	if unhealthy {
		s.HealthStatus = ApplicationLiveStateSnapshot_UNHEALTHY
	}
	if unknown {
		s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
	}
}
```
ok, let's me suggest a clear version of what you both trying to do 😄

Reply:
```suggestion
func (s *ApplicationLiveStateSnapshot) DetermineApplicationHealthStatus() {
	app := s.ApplicationLiveState
	if app == nil {
		return
	}
	
	s.HealthStatus = ApplicationLiveStateSnapshot_HEALTHY

	for _, r := range app.Resources {
		if r.HealthStatus == ResourceState_UNHEALTHY {
			s.HealthStatus = ApplicationLiveStateSnapshot_UNHEALTHY
			return
		}
	}

	for _, r := range app.Resources {
		if r.HealthStatus == ResourceState_UNKNOWN {
			s.HealthStatus = ApplicationLiveStateSnapshot_UNKNOWN
			return
		}
	}
}
```
Or another version at least could be this to avoid miss set health status

Reply:
Thanks both @khanhtc1202 @t-kikuc 
I fixed it to reduce one for statement.
[07acfbc](https://github.com/pipe-cd/pipecd/pull/5454/commits/07acfbc24598d70e989405dacece1eab76443025)